provider/database_db_homes_resource.go

```diff
@@ -0,0 +1,243 @@
+}
+
+func updateDBHome(d *schema.ResourceData, m interface{}) (e error) {
+	sync := &DBHomeResourceCrud{}
+	sync.D = d
+	sync.Client = m.(*OracleClients).client
+	return crud.UpdateResource(d, sync)
+}
+
+func deleteDBHome(d *schema.ResourceData, m interface{}) (e error) {
+	sync := &DBHomeResourceCrud{}
+	sync.D = d
+	sync.Client = m.(*OracleClients).clientWithoutNotFoundRetries
+	return crud.DeleteResource(d, sync)
+}
+
+type DBHomeResourceCrud struct {
+	crud.BaseCrud
+	Res *baremetal.DBHome
+}
+
+func (s *DBHomeResourceCrud) ID() string {
+	return s.Res.ID
+}
+
+func (s *DBHomeResourceCrud) CreatedPending() []string {
+	return []string{baremetal.ResourceProvisioning}
+}
+
+func (s *DBHomeResourceCrud) CreatedTarget() []string {
+	return []string{baremetal.ResourceAvailable}
+}
+
+func (s *DBHomeResourceCrud) DeletedPending() []string {
+	return []string{baremetal.ResourceTerminating}
+}
+
+func (s *DBHomeResourceCrud) DeletedTarget() []string {
+	return []string{baremetal.ResourceTerminated}
+}
+
+func (s *DBHomeResourceCrud) Create() (e error) {
+	opts := &baremetal.CreateDBHomeOptions{}
+	databaseRaw := s.D.Get("database").([]interface{})
+	database := &baremetal.CreateDatabaseDetails{}
+	if len(databaseRaw) > 0 {
+		databaseMap := databaseRaw[0].(map[string]interface{})
+		adminPassword := databaseMap["admin_password"].(string)
+		database.AdminPassword = adminPassword
+		dbName := databaseMap["db_name"].(string)
+		database.DBName = dbName
+		characterSet, ok := databaseMap["character_set"]
+		if ok && characterSet != nil {
+			database.CharacterSet = characterSet.(string)
+		}
+		dbWorkload, ok := databaseMap["db_workload"]
+		if ok && dbWorkload != nil {
+			database.DBWorkload = dbWorkload.(string)
+		}
+		ncharacterSet, ok := databaseMap["ncharacter_set"]
+		if ok && ncharacterSet != nil {
+			database.NcharacterSet = ncharacterSet.(string)
+		}
+		pdbName, ok := databaseMap["pdb_name"]
+		if ok && pdbName != nil {
+			database.PDBName = pdbName.(string)
+		}
+
+	}
+	dbSystemID := s.D.Get("db_system_id").(string)
+	dbVersion := s.D.Get("db_version").(string)
+	displayName, ok := s.D.GetOk("display_name")
+	if ok && displayName != nil {
+		opts.DisplayName = displayName.(string)
+	}
+
+	s.Res, e = s.Client.CreateDBHome(database, dbSystemID, dbVersion, opts)
+	return
+}
+
+func (s *DBHomeResourceCrud) Get() (e error) {
+	dbHomeID := s.D.Get("id").(string)
+
+	s.Res, e = s.Client.GetDBHome(dbHomeID)
+	return
+}
+
+func (s *DBHomeResourceCrud) Update() (e error) {
+	opts := &baremetal.UpdateDBHomeOptions{}
+	dbHomeID := s.D.Get("id").(string)
+	dbVersion, ok := s.D.GetOk("db_version")
+	if ok && dbVersion != nil {
+		opts.DBVersion = dbVersion.(string)
+	}
+
+	s.Res, e = s.Client.UpdateDBHome(dbHomeID, opts)
+	return
+}
+
+func (s *DBHomeResourceCrud) Delete() (e error) {
+	dbHomeID := s.D.Get("id").(string)
+
+	e = s.Client.DeleteDBHome(dbHomeID, nil)
+	return
+}
+
+func (s *DBHomeResourceCrud) SetData() {
```

Comment:
See https://github.com/oracle/terraform-provider-oci/blob/79fc22feeafb60a5951a8cc3c2c270615316e548/provider/core_instance_resource.go#L361 for an example of updating a nested resource. (It's the only example I'm aware of.) That also updates the nested resource.

Reply:
change needs to be made in the generator